src/storage/mvcc/txn.rs

```diff
@@ -1032,6 +1039,61 @@ impl<S: Snapshot, P: PdClient + 'static> MvccTxn<S, P> {
         }
     }
 
+    /// Check the status of a secondary (optimistic) lock.
+    ///
+    /// It checks whether the given secondary lock exists. If the lock exists,
+    /// the lock information is returned. Otherwise, it searches the write CF
+    /// for the commit record of the lock and returns the commit timestamp
+    /// (0 if the lock is not committed).
+    ///
+    /// If the lock does not exist or is a pessimistic lock, to prevent the
+    /// status being changed, a rollback may be written and this rollback
+    /// needs to be protected.
+    pub fn check_secondary_lock(
+        &mut self,
+        key: &Key,
+        start_ts: TimeStamp,
+    ) -> Result<(SecondaryLockStatus, Option<ReleasedLock>)> {
+        let mut released_lock = None;
+        let (status, need_rollback) = match self.reader.load_lock(&key)? {
+            Some(lock) if lock.ts == start_ts => {
+                let status = if lock.lock_type == LockType::Pessimistic {
+                    released_lock = self.unlock_key(key.clone(), true);
+                    SecondaryLockStatus::RolledBack
+                } else {
+                    SecondaryLockStatus::Locked(lock)
```

Comment:
In this branch, if the corresponding prewrite lock exists, `need_rollback` flag should be false? Maybe we could add test cases to check flags or rollback records.

Reply:
...sorry 😷
I wrote a wrong assertion in the test... I've fixed it and add assertions after each `check_secondary` operation.